postgres/datadog_checks/postgres/connections.py

```diff
@@ -143,17 +150,25 @@ def get_connection(self, dbname: str, ttl_ms: int, timeout: int = None, persiste
         Note that leaving a connection context here does NOT close the connection in psycopg2;
         connections must be manually closed by `close_all_connections()`.
         """
+        interrupted_error_seen = False
+
+        with self._mu:
+            db = self._get_connection_raw(dbname, ttl_ms, timeout, startup_fn, persistent)
         try:
-            with self._mu:
-                db = self._get_connection_raw(dbname, ttl_ms, timeout, persistent)
             yield db
+        except InterruptedError:
```

Comment:
I think we want to handle an `InterfaceError`, right? 

these were the errors i was seeing 
```
    with self.db.cursor() as cursor:
psycopg2.InterfaceError: connection already closed
"}]
```

and 

```
  File \"/opt/datadog-agent/embedded/lib/python3.9/site-packages/datadog_checks/postgres/postgres.py\", line 385, in _run_query_scope
    cursor.execute(formatted_query)
psycopg2.InterfaceError: cursor already closed
"}]
```